src/game_manager.rs

```diff
@@ -84,20 +87,20 @@ impl GameManager {
         match self.fsm {
             GameFsm::Combat => {
                 // TODO(orkaboy): actually handle combat. For now, mash!
-                if self.btn.update(&mut context.gamepad, dt) {
-                    self.btn = ButtonPress {
-                        action: SosAction::Confirm,
-                        press_time: 0.1,
-                        release_time: 0.2,
-                        ..Default::default()
-                    };
-                }
 
                 if !cmd.encounter_active {
                     context.gamepad.release_all();
                     self.fsm = GameFsm::Route;
                     // Signal return to sequencer
                     self.sequencer.on_event(context, &GameEvent::Combat);
+                } else if let Some(combat) = self.combat_manager.as_mut() {
+                    if combat.update(context, dt) {
+                        self.combat_manager = None;
+                        self.fsm = GameFsm::Route;
+                    }
+                } else {
+                    context.gamepad.release_all();
+                    self.combat_manager = Some(CombatManager::default());
                 }
```

Comment:
Won't the top if branch here potentially prevent the combat manager from being cleared? Not sure if this is desired behavior or not.

Reply:
I would say its not entirely necessary either way, but I think having a clean slate is the most sane option - happy to eat the allocation cost to avoid bugs.